lib/workers/repository/process/write.ts

```diff
@@ -39,11 +63,37 @@ export async function writeUpdates(
       meta['baseBranch'] = baseBranch;
     }
     addMeta(meta);
-    const branchExisted = branchExists(branch.branchName);
+    const branchExisted = branchExists(branchName);
+    let branchCache = {} as BranchCache;
+    if (branchExisted && config.repositoryCache === 'enabled') {
+      branchCache =
+        cachedBranches?.find((br) => br.branchName === branchName) ??
+        ({} as BranchCache);
+
+      if (Object.keys(branchCache).length === 0) {
+        logger.debug(`No branch cache found for ${branch.branchName}`);
+      }
+    }
+    const branchManagersFingerprint = hasha(
+      branch.upgrades
+        .map((upgrade) => hashMap.get(upgrade.manager) ?? upgrade.manager)
+        .filter(is.string)
+    );
+    branch.branchFingerprint = hasha([
+      JSON.stringify(branch),
+      branchManagersFingerprint,
+    ]);
+    branch.skipBranchUpdate = canSkipBranchUpdateCheck(
+      branchCache,
+      branch.branchFingerprint
+    );
     const res = await processBranch(branch);
     branch.prBlockedBy = res?.prBlockedBy;
     branch.prNo = res?.prNo;
     branch.result = res?.result;
+    if (!(res?.branchExists && res?.updateBranchFingerprint)) {
```

Comment:
This logic line is confusing to read due to the negation. Also, does it handle the case where a branch is *created*?

Reply:
Sorry for that couldn't think of a better way that time. I think the `branchExists` check is unnecessary cause we only pass the `updateBranchFingerprint` after we have commitFilesToBranch and its true only if the commitSha is not null. So it should be enough info to decide of we can update cached fingerprint.